### PR TITLE
follow xdg base dir specification

### PR DESCRIPTION
**Pull request by haseebeqx**
implements #675

### CODE REVIEW

**Comment by haseebeqx** [comment]
@ericmj I created a draft implementation. Please confirm whether it is in the direction you want. After that I will finish this.

File: lib/hex/config.ex [hunk 1]
```diff
@@ -49,12 +49,20 @@ defmodule Hex.Config do
...
-      Path.expand(System.get_env("HEX_HOME") || "~/.hex")
+      find_config_home()
     end
   end
 
+  def find_config_home() do
+    case {System.get_env("HEX_HOME"), System.get_env("XDG_CONFIG_HOME")} do
+        {directory, _} when is_binary(directory) -> Path.expand(directory)
+        {nil, directory} when is_binary(directory) -> Path.join(Path.expand(directory), "hex")
+        {nil, nil} -> "~/.hex"
```
Comment (ericmj) [inline lib/hex/config.ex]:
```suggestion
        {nil, nil} -> Path.expand("~/.hex")
```

File: lib/hex/state.ex [hunk 2]
```diff
@@ -62,8 +72,8 @@ defmodule Hex.State do
       config: [:http_timeout],
       fun: {__MODULE__, :to_integer}
     },
-    home: %{
-      env: ["HEX_HOME"],
+    data_home: %{
+      env_path_join: [{"HEX_HOME", ""},{"XDG_DATA_HOME", "hex"}],
```
Comment (ericmj) [inline lib/hex/state.ex]:
```suggestion
      env_path_join: [{"HEX_HOME", ""}, {"XDG_DATA_HOME", "hex"}],
```

**Comment by haseebeqx** [comment]
Thanks @ericmj 
I will continue this PR now.

**Comment by haseebeqx** [comment]
@ericmj Fixed failing tests. Please review.

**Comment by haseebeqx** [comment]
Is there anything else needed in this?

**Comment by wojtekmach** [comment]
Sorry for the delay, this is great, thank you so much!

**Comment by wojtekmach** [comment]
we also need to support this:

```
% mix hex.docs offline ecto
Docs fetched: /Users/wojtek/.hex/docs/hexpm/ecto/3.3.4
```

which directory is the most appropriate? `XDG_CACHE_HOME`? `XDG_DATA_HOME`?

**Comment by wojtekmach** [comment]
oh, it's already supported, please ignore me.

**Comment by haseebeqx** [comment]
💚❤️💚❤️💚❤️💚❤️💚❤️

**Pull request merged**